store/cache.go

```diff
@@ -79,3 +91,49 @@ func (cache cache) Entries() int {
 
 	return len(cache.data)
 }
+
+func (cache cache) Iterator() KVStoreIterator {
+	cache.mu.RLock()
+	defer cache.mu.RUnlock()
+
+	return newCacheIterator(cache.data)
+}
+
+type cacheIterator struct {
+	index  int
+	keys   []string
+	values [][]byte
+}
+
+func newCacheIterator(data map[string][]byte) *cacheIterator {
```

Comment:
Iterators can be quite dangerous; they can allow for users of the cache to accidentally subvert needing mutex-protected access to the underlying data if they are not implemented correctly. Be aware that the thing "saving" us here is that the underlying keys/values are not accessed from the iterator directly and instead unmarshaling is used. This is fine, but it should be documented so that someone making modifications to this cache will not accidentally mess it up.